这个自动打包对于linux用户真的头疼，更新又勤，又忍不住去更新

很多次都遇到包有问题了 
新包：
  - 包元数据声明为 Architecture: amd64
  - 主程序 /tmp/netcatty-debcheck.8iQm91/opt/Netcatty/netcatty 是 x86-64
  - 但关键依赖 /tmp/netcatty-debcheck.8iQm91/opt/Netcatty/resources/
    app.asar.unpacked/node_modules/node-pty/build/Release/pty.node 却是 ARM
    aarch64
真的头疼

lhuanyun@deianm:~/桌面$ ./Netcatty-1.0.62-linux-x86_64.AppImage Uncaught exception: Error: Failed to load native module: pty.node, checked: build/Release, build/Debug, prebuilds/linux-x64: Error: Cannot find module './prebuilds/linux-x64//pty.node' Require stack: - /tmp/.mount_Netcat2w7kUt/resources/app.asar/node_modules/node-pty/lib/utils.js - /tmp/.mount_Netcat2w7kUt/resources/app.asar/node_modules/node-pty/lib/index.js - /tmp/.mount_Netcat2w7kUt/resources/app.asar/electron/bridges/terminalBridge.cjs - /tmp/.mount_Netcat2w7kUt/resources/app.asar/electron/main.cjs - at Object.loadNativeModule (/tmp/.mount_Netcat2w7kUt/resources/app.asar/node_modules/node-pty/lib/utils.js:36:11) at Object.<anonymous> (/tmp/.mount_Netcat2w7kUt/resources/app.asar/node_modules/node-pty/lib/unixTerminal.js:27:22) at Module._compile (node:internal/modules/cjs/loader:1769:14) at Module._extensions..js (node:internal/modules/cjs/loader:1902:10) at Module.load (node:internal/modules/cjs/loader:1488:32) at Module._load (node:internal/modules/cjs/loader:1307:12) at c._load (node:electron/js2c/node_init:2:17999) at TracingChannel.traceSync (node:diagnostics_channel:328:14) at wrapModuleLoad (node:internal/modules/cjs/loader:252:24) at Module.require (node:internal/modules/cjs/loader:1511:12) /tmp/.mount_Netcat2w7kUt/resources/app.asar/electron/main.cjs:28 throw err; ^ Error: Failed to load native module: pty.node, checked: build/Release, build/Debug, prebuilds/linux-x64: Error: Cannot find module './prebuilds/linux-x64//pty.node' Require stack: - /tmp/.mount_Netcat2w7kUt/resources/app.asar/node_modules/node-pty/lib/utils.js - /tmp/.mount_Netcat2w7kUt/resources/app.asar/node_modules/node-pty/lib/index.js - /tmp/.mount_Netcat2w7kUt/resources/app.asar/electron/bridges/terminalBridge.cjs - /tmp/.mount_Netcat2w7kUt/resources/app.asar/electron/main.cjs - at Object.loadNativeModule (/tmp/.mount_Netcat2w7kUt/resources/app.asar/node_modules/node-pty/lib/utils.js:36:11) at Object.<anonymous> (/tmp/.mount_Netcat2w7kUt/resources/app.asar/node_modules/node-pty/lib/unixTerminal.js:27:22) at Module._compile (node:internal/modules/cjs/loader:1769:14) at Module._extensions..js (node:internal/modules/cjs/loader:1902:10) at Module.load (node:internal/modules/cjs/loader:1488:32) at Module._load (node:internal/modules/cjs/loader:1307:12) at c._load (node:electron/js2c/node_init:2:17999) at TracingChannel.traceSync (node:diagnostics_channel:328:14) at wrapModuleLoad (node:internal/modules/cjs/loader:252:24) at Module.require (node:internal/modules/cjs/loader:1511:12) Node.js v24.11.1

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -122,10 +122,13 @@ jobs:
           npm pkg set version="${VERSION}"
 
       - name: Prepare node-pty Linux runtime
+        env:
+          npm_config_arch: x64
         run: bash scripts/ensure-node-pty-linux.sh prepare x64
 
       - name: Build package
         env:
+          npm_config_arch: x64
           ELECTRON_BUILDER_PUBLISH: "never"
         run: npm run pack:linux-x64
 

--- a/electron-builder.config.cjs
+++ b/electron-builder.config.cjs
@@ -6,6 +6,7 @@ module.exports = {
     productName: 'Netcatty',
     artifactName: '${productName}-${version}-${os}-${arch}.${ext}',
     icon: 'public/icon.png',
+    npmRebuild: false,
     directories: {
         buildResources: 'build',
         output: 'release'

--- a/scripts/ensure-node-pty-linux.sh
+++ b/scripts/ensure-node-pty-linux.sh
@@ -61,7 +61,7 @@ prepare() {
 
   echo "[node-pty] rebuilding native modules for Electron on linux-${arch}"
   log_electron_runtime_info
-  npx electron-rebuild
+  npx electron-rebuild --arch "${arch}"
 
   test -f "${release_dir}/pty.node"
 

